src/containers/CreateCommunity.js

```diff
@@ -0,0 +1,434 @@
+
+        if (some(error)) {
+          this.resetValidation('slug')
+        } else {
+          return this.checkUnique('slug', value)
+        }
+        break
+      case 'beta_access_code':
+        this.setError({codeBlank: !value})
+
+        if (!value) {
+          this.resetValidation('beta_access_code')
+        } else if (includes(value, '/')) {
+          this.setError({codeInvalid: true})
+        } else {
+          return this.checkUnique('beta_access_code', value)
+        }
+        break
+    }
+  }
+
+  attachImage (type) {
+    let { dispatch } = this.props
+    let community = {id: 'new', slug: 'new'}
+    switch (type) {
+      case 'avatar':
+        dispatch(uploadImage(avatarUploadSettings(community)))
+        break
+      case 'banner':
+        dispatch(uploadImage(bannerUploadSettings(community)))
+    }
+    return false
+  }
+
+  validateAll () {
+    let { community } = this.props
+    return Promise.all(
+      ['name', 'description', 'slug', 'beta_access_code', 'category']
+      .map(key => this.validate(key, community[key]))
+    )
+  }
+
+  submit = () => {
+    let { validating, dispatch, community, currentUser } = this.props
+    if (validating) return
+
+    this.validateAll().then(() => {
+      if (some(id => id, this.props.errors)) return scrollToBottom()
+
+      dispatch(createCommunity(community))
+      .then(({ payload }) => {
+        if (some(id => id, this.props.errors)) {
+          return scrollToBottom()
+        } else {
+          trackEvent(ADDED_COMMUNITY, {community})
+          saveCurrentCommunityId(dispatch, payload.community_id, currentUser.id)
+          dispatch(navigate(`/create/invite`))
+        }
+      })
+    })
+  }
+
+  render () {
+    const { community, errors } = this.props
+
+    const { expanded } = this.state
+
+    return <Modal title='Create your community.'
+      className='create-community-one'
+      subtitle="Let's get started unlocking the creative potential of your community with Hylo"
+      standalone>
+        <ModalInput label='Name' ref='name' onChange={this.set('name')}/>
+        {errors.nameBlank && <p className='help error'>Please fill in this field.</p>}
+        {errors.nameUsed && <p className='help error'>This name is already in use.</p>}
+        <ModalInput label='URL' ref='url' prefix='https://hylo.com/c/' onChange={this.set('slug')}/>
+        {errors.slugBlank && <p className='help error'>Please fill in this field.</p>}
+        {errors.slugInvalid && <p className='help error'>Use lowercase letters, numbers, and hyphens only.</p>}
+        {errors.slugUsed && <p className='help error'>This URL is already in use.</p>}
+        <ModalInput label='Description' ref='description' type='textarea' onChange={this.set('description')}/>
+        {errors.descriptionBlank && <p className='help error'>Please fill in this field.</p>}
+        <ModalInput label='Invitation Code' ref='beta_access_code' onChange={this.set('beta_access_code')}/>
```

Comment:
Maybe we skip setting up the code on the first screen? @edwardwest what do you think? this would be a good way to test phasing it out.